specification/resources/resource-manager/Microsoft.Subscription/2017-11-01-preview/subscriptionDefinitions.json

```diff
@@ -69,7 +69,11 @@
             "$ref": "#/parameters/scopeParameter"
           },
           {
-            "$ref": "#/parameters/subscriptionDefinitionNameParameter"
+            "name": "subscriptionDefinitionName",
```

Comment:
curious... why did you decide on placing the parameter definition inline, instead of the reference?

Reply:
Because it affects the client sdks including cli and python.  It made this extra parameters always required rather than parameters just for the methods that they are inlined for .

Reply:
Ok, I'm fine with the inline. If you wanted to make method parameters instead of client paramaters, you can use https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-parameter-location on your shared parameter definition.